applications/GeoMechanicsApplication/custom_utilities/equation_of_motion_utilities.cpp

```diff
@@ -0,0 +1,60 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+// Project includes
+
+// Application includes
+#include "custom_utilities/equation_of_motion_utilities.h"
+
+namespace Kratos
+{
+
+Matrix GeoEquationOfMotionUtilities::CalculateMassMatrix(SizeType      dimension,
+                                                         SizeType      number_U_nodes,
+                                                         SizeType      NumberIntegrationPoints,
+                                                         const Matrix& Nu_container,
+                                                         const Vector& rSolidDensities,
+                                                         const std::vector<double>& rIntegrationCoefficients)
+{
+    const SizeType block_element_size = number_U_nodes * dimension;
+    Matrix         Nu                 = ZeroMatrix(dimension, block_element_size);
+    Matrix         aux_density_matrix = ZeroMatrix(dimension, block_element_size);
+    Matrix         density_matrix     = ZeroMatrix(dimension, dimension);
+    Matrix         mass_matrix        = ZeroMatrix(block_element_size, block_element_size);
+
+    for (unsigned int g_point = 0; g_point < NumberIntegrationPoints; ++g_point) {
+        GeoElementUtilities::AssembleDensityMatrix(density_matrix, rSolidDensities(g_point));
+        GeoElementUtilities::CalculateNuMatrix(dimension, number_U_nodes, Nu, Nu_container, g_point);
+        noalias(aux_density_matrix) = prod(density_matrix, Nu);
+        mass_matrix += prod(trans(Nu), aux_density_matrix) * rIntegrationCoefficients[g_point];
+    }
+    return mass_matrix;
+}
```

Comment:
This became a very easy-to-read and generic function!

Reply:
Thank you, it is a real team work!